doc/devel/style_guide.rst

```diff
@@ -0,0 +1,268 @@
+
+===========
+Style Guide
+===========
+
+This guide is for contributors to understand conventions and expectations for
+quality control of Matplotlib written content.
+
+.. seealso::
+
+    For more information about contributing, see the :ref:`documenting-matplotlib`
+    section.
+
+Expository Language
+===================
+
+For explanatory writing, the following guidelines are for clear and concise
+language use.
+
+Terminology
+-----------
+
+There are several key terms in Matplotlib that are standards for 
+reliability and consistency in documentation. They are case-sensitive and are
```

Comment:
To be discussed with other devs:

AFAIK, we haven't discussed that. Up to now, I use a slightly different approach: When refrerring to the general concept, I use lower-case:

> `plt.show()` displays all figures.

I use the uppercase version when the code representation (i.e. the class) of that concept is important.

> `plt.figure()` returns a `.Figure`.

But then I usually use it as ReST reference. I rarely ever use upper-case "Figure" in plain text.

This approach is a little less formal and is motivated in a more narrative natural language appearance: If the exact class is not relevant, don't put emphasis on it, because that would impact reading flow. Of course this is not unproblematic as well. "figure" is still a technical term with a specific meaning in our context. But every domain language has technical terms and they are not specifically highlighted. For example, when a physicist talks about "force", that is an exactly defined concept, but we would never highlight that in written texts.

Partly we rely on the fact that the general meaning of a term is not too far away from our term and is understandable without knowing the code details.

This approach comes to a limit with `Axes` (#18726), but I would be happy to only work around this special case.

While I have a slight preference for the more informal approach, I also see the benefit of being more explicit. Either way, it's worth discussing the advantages and disadvantages.

@jeromefv I'd welcome if you can comment on this from the technical writer perspective as well.

Reply:
Those are valid points! I think the reStructuredText reference is a huge help. I'd be for using that way of referencing the specific concept for respective terms.

One of the biggest hangups with allowing a lowercase term or giving flexibility with casing is the lack of consistency. Whether that's `figure` or `Figure` or [.Figure](https://matplotlib.org/stable/api/_as_gen/matplotlib.figure.Figure.html?highlight=figure#matplotlib.figure.Figure), if all three are accepted, there should be a place in the docs that states that. In that sense, there isn't much of an "incorrect" way of using the term, save for `Axes` in Matplotlib's case.

For me as a technical writer, I think when a term is easy to remember and specific, it'll be used more often. That makes it easier for people to understand and easier to share. However, I know it can be more work to drop a reStructuredText reference for every instance of a term, so then it's less likely to get used. Using a capital letter for those specific terms is a compromise in my eyes.

Ultimately, the way that people talk about the terms is the one that catches on. If folks are interchanging the case for terms and folks aren't confused, then that's how the terms are used and to have that documented in a single place is a good thing. I hope that isn't too back and forth!

Reply:
Consensus for the call this week:

  - `.Figure` not more than once is a block
  - Figure when you want to indicate that the sentence is about _our_ Figure classes specifically e.g. "In Matplotlib there can be several Axes in a Figure".
  - figure when the sentence is very generic "Frequently, a figure contains both an image and a colorbar"

Reply:
@tacaswell Sounds good to me! I'll work on the Style Guide and incorporate those changes before Monday 3/1. Thank you and all the devs!